### PR TITLE
Fix #104: Unsupported links

### LINKED ISSUE #104: Hide unsupported libraries
Logic is:

1. Library is not marked as "starred" and last update was about year ago - mark library as "unsupported".
2. On UI hide unsupported libraries by default, show them with mark in search results/full view

### PR DESCRIPTION
**Pull request by madhead**
Make sure that you are making pull request against [`main`](https://github.com/KotlinBy/awesome-kotlin/tree/main) branch. Pull requests against `readme` branch will not be accepted, because all manual changes to this branch will be overridden by changes from `main` branch. Consult [contributing.md](https://github.com/KotlinBy/awesome-kotlin/blob/main/.github/contributing.md) for details.

Checklist: 

- [x] Is this pull request against the branch `main`?

### CODE REVIEW

File: src/main/kotlin/link/kotlin/scripts/LinksProcessor.kt [hunk 1]
```diff
@@ -23,7 +26,9 @@ private class DefaultLinksProcessor(
     private val mapper: ObjectMapper,
     private val httpClient: HttpClient,
     private val linksChecker: LinksChecker
-): LinksProcessor {
+) : LinksProcessor {
+    val now: Instant = Instant.now()
```
Comment (IRus) [inline src/main/kotlin/link/kotlin/scripts/LinksProcessor.kt:30]: private?

File: src/main/kotlin/link/kotlin/scripts/model/Link.kt [hunk 2]
```diff
@@ -13,5 +13,6 @@ data class Link(
     val tags: List<String> = emptyList(),
     val star: Int? = null,
     val update: String? = null,
-    val archived: Boolean = false
+    val archived: Boolean = false,
+    val unsupported: Boolean? = false,
```
Comment (IRus) [inline src/main/kotlin/link/kotlin/scripts/model/Link.kt:17]: I think can be just `Boolean`

**Review by IRus — CHANGES_REQUESTED** [review]
Nice implementation, thanks for better naming. I have just two minor comments

File: src/main/kotlin/link/kotlin/scripts/LinksProcessor.kt (hunk 1 above)
Comment (madhead) [reply src/main/kotlin/link/kotlin/scripts/LinksProcessor.kt]: Done

File: src/main/kotlin/link/kotlin/scripts/model/Link.kt (hunk 2 above)
Comment (madhead) [reply src/main/kotlin/link/kotlin/scripts/model/Link.kt]: Done

**Pull request merged**